pkg/asset/machines/master.go

```diff
@@ -107,6 +108,9 @@ const (
 	// masterUserDataFileName is the filename used for the master
 	// user-data secret.
 	masterUserDataFileName = "99_openshift-cluster-api_master-user-data-secret.yaml"
+
+	// masterUserDataFileName is the filename used for the control plane machine sets.
+	controlPlaneMachineSetFileName = "99_openshift-machine-api_master-control-plane-machine-set.yaml"
```

Comment:
Do we know if it's possible to make sure that the machines are created before the CPMS resource is created rather than the other way around? If they are created in the wrong order then we may end up with too many replicas, in which case the CPMS will go degraded

Reply:
I don't believe we have a mechanism to control this... 
This CPMS resource is going to be applied by the bootstrap node, while (ATM) the control plane nodes are going to be provisioned independently by installer/terraform...

Reply:
I think we should actually be ok, we've implemented some checks in the CPMSO now which should prevent anything happening until the initial set of Control Plane Machines are ready. I'll rerun the tests and see what happens

Reply:
So, you (Joel) mentioned that we should discuss in the arch call having mao-provisioned control plane nodes. That may perhaps be the most straightforward way of handling this... I'm assuming that solution would be that we run the MAO on the bootstrap node.

> I don't believe we have a mechanism to control this...
> This CPMS resource is going to be applied by the bootstrap node,

Let me double check this. I'm a bit shaky on how the manifests are applied on the bootstrap node. I'm pretty sure these manifests do not depend on running control plane nodes, but want to confirm.

Reply:
> That may perhaps be the most straightforward way of handling this... I'm assuming that solution would be that we run the MAO on the bootstrap node.

I suspect so. Not sure what that would look like but we would need MAO and CPMSO and their friends to be running to be able to provision. We'd also want to add a bootstrap mode to CPMSO to allow it to go past some of the safety checks it currently has, but we can expand that conversation on thursday

Reply:
It looks like I was incorrect anyway, the "bootstrap manifests" are applied by cluster bootstrap, which only runs after etcd (and therefore control plane nodes) are up.